distributed/protocol/pickle.py

```diff
@@ -0,0 +1,40 @@
+from __future__ import print_function, division, absolute_import
+
+import pickle
+
+import cloudpickle
+
+
+def dumps(x):
+    """ Manage between cloudpickle and pickle
+
+    1.  Try pickle
+    2.  If it is short then check if it contains __main__
+    3.  If it is long, then first check type, then check __main__
+    """
+    try:
+        result = pickle.dumps(x, protocol=pickle.HIGHEST_PROTOCOL)
+        if len(result) < 1000:
+            if b'__main__' in result:
+                return cloudpickle.dumps(x, protocol=pickle.HIGHEST_PROTOCOL)
+            else:
+                return result
+        else:
+            if isinstance(x, pickle_types) or b'__main__' not in result:
```

Comment:
Originally these were types for which pickle definitely worked and where we didn't want to use cloudpickle.  This notably included numpy arrays.  I must have lost the variable while moving around code.  Clearly this code isn't covered though and needs to be fixed.